doc/dev/kdb-operations.md

```diff
@@ -22,33 +24,32 @@ The basic flow of this operation is:
 
 1. Create empty `KDB` instance
 2. Configure `KDB` instance for bootstrap
-3. Run bootstrap `get` operation
-4. Setup default global plugins <!-- TODO: maybe change name; global plugins are very different from normal plugins (although the C API is the same)-->
-5. Process contract
-6. Parse mountpoints
-7. Configure `KDB` instance with real mountpoints
-8. Add hard coded mountpoints to `KDB` instance
-
-> **Note:** Despite sharing a name, the `open` operation **does not** call the `elektra<Plugin>Open` function of any plugin (except within the bootstrap `get` operation).
+3. Run bootstrap `get` operation:
+   This loads the contents of `system:/elektra/mountpoints` so that the mountpoints can be configured.
+4. Process contract and set up plugins for hooks (see [Hooks](hooks.md))
+5. Parse mountpoints:
+   This transforms the contents of `system:/elektra/mountpoints` into the internal state stored in a `KDB` instance.
+6. Reconfigure `KDB` with real mountpoints:
+   This switches the `KDB` instance from bootstrap mode to use the real mountpoint state created above.
+7. Add hardcoded mountpoints to `KDB` instance:
+   There are a few hardcoded mountpoints (root mountpoints, `system:/elektra/modules`, `system:/elektra/version`, etc.) that are always present.
+   They are added in this step.
 
 Namespaces in mountpoint configs:
 
-- cascading mountpoints act as an alias for identical `proc:/`, `dir:/`, `user:/`, `system:/` and `default:/` mountpoints.
-- `meta:/` mountpoints are illegal
-- explicit `default:/` mountpoints are illegal
-  They can only exist via cascading mountpoints.
-  This is so that default values can be checked and validated by plugins.
-- `spec:/` mountpoints may exist, but they are treated differently during `get` and `set`
-- `proc:/` mountpoints may exist, but they are read-only (see below)
-- `dir:/`, `user:/` and `system:/` mountpoints can be created without restrictions, except for the one below
+- `dir:/`, `user:/` and `system:/` mountpoints can be created without restrictions, except for the reserved sections listed below.
+- `spec:/` mountpoints can be created with the same restrictions, but they are also treated specially during `get` and `set`.
+- `proc:/` mountpoints are always read-only and receive special treatment during `get`
+- `default:/` mountpoints are read-only and receive special treatment during `get`, specifically they only go through the `poststorage` phase
```

Comment:
It would be interesting what the special treatment is.

Reply:
Such forward references are not ideal, even less ideal if they are implicit.

Reply:
I'll add an "explained below", but IMO it is very obvious where to look already. The file clearly has one section per operations, so if I refer to the `get` operation that's where you should look.

I also know that forward references are not ideal, but there is no way around it here. Explaining all of `get` before `open` doesn't make sense. Explaining just the special treatment here is also not ideal, because that should clearly be explained in `get`. Duplicating the information will just lead to conflicting information in future.